src/main/java/com/fasterxml/jackson/core/json/ReaderBasedJsonParser.java

```diff
@@ -1848,9 +1848,11 @@ protected final String _parseName() throws IOException
             int ch = _inputBuffer[ptr];
             if (ch < codes.length && codes[ch] != 0) {
                 if (ch == '"') {
-                    int start = _inputPtr;
+                    final int start = _inputPtr;
                     _inputPtr = ptr+1; // to skip the quote
-                    return _symbols.findSymbol(_inputBuffer, start, ptr - start, hash);
+                    final int len = ptr - start;
+                    _streamReadConstraints.validateNameLength(len);
```

Comment:
Ideally we should only (have to) check length when adding new name in `CharsToNameCanonicalizer` (and byte-based counterpart); not for every time we decode a name.

But to do that would need to pass constraints... that might be doable in `makeChild()` method. Maybe adding new variant that takes `JsonFactory`, instead of flags.